src/pytorch_lightning/strategies/ipu.py

```diff
@@ -162,6 +162,7 @@ def setup(self, trainer: "pl.Trainer") -> None:
             if self.lightning_module.trainer.enable_validation:
                 model = poptorch.inferenceModel(model=model, options=inference_opts)
                 self.poptorch_models[RunningStage.VALIDATING] = model
+                self.poptorch_models[RunningStage.SANITY_CHECKING] = model
```

Comment:
So, I'm not 100 % sure I'd call it a bug. But the piece of this PR was trying to get `self.replication_factor`, which would fail in sanity checking stage, since we already had a trainer connected, but didn't have the model saved.